test/api.jl

```diff
@@ -115,5 +116,30 @@ recommendation = provide_recommendation(trial_vector)
 recommendation_multi = provide_recommendation(trial_vector_multihyperparams)
 @test recommendation_multi == Dict(:x => 3, :y => 7)
 
+# Test case reproducing the issue in “obs_memo” function from the higher level
+space = Dict(
+    :a => HP.Choice(:a, [8, 9, 10]),
+    :b => HP.Choice(:b, [1, 2, 3])
+    )
+
+Random.seed!(10)
+trials = TreeParzen.Trials.Trial[]
+
+for i in 1:30
+    trial = ask(space, trials, TreeParzen.Config())
+    println("Iteration: ", i, ", ", trial.hyperparams)
```

Comment:
Please remove the print

Reply:
This branch will not be merged, which is only used for developers to check all the iteration results to fully understand the bug. When the bug is fixed, a finally branch will be pushed without this print.

Reply:
This branch **should** be merged